dependency/actions/compile/entrypoint/main.go

```diff
@@ -0,0 +1,480 @@
+		CommandWithArgs{"make", []string{"install"}},
+	}
+
+	for _, command := range commands {
+		err = runCommand(dir, command.Command, command.Args...)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func installExtensions(dependencyDir string) error {
+	var extensions []ExtensionsData
+	jsonBlob, _ := os.ReadFile("/tmp/entrypoint/extensions/extensions-8.1.json")
+	err := json.Unmarshal(jsonBlob, &extensions)
+	if err != nil {
+		return err
+	}
+
+	idx := slices.IndexFunc(extensions, func(e ExtensionsData) bool { return e.Name == "rabbitmq-c" })
+	err = installRabbitMQC(extensions[idx])
+	if err != nil {
+		return err
+	}
+
+	idx = slices.IndexFunc(extensions, func(e ExtensionsData) bool { return e.Name == "lua" })
+	err = installLua(extensions[idx])
+	if err != nil {
+		return err
+	}
+
+	idx = slices.IndexFunc(extensions, func(e ExtensionsData) bool { return e.Name == "hiredis" })
+  err = installHiRedis(extensions[idx])
+	if err != nil {
+		return err
+	}
+
+	idx = slices.IndexFunc(extensions, func(e ExtensionsData) bool { return e.Name == "librdkafka" })
+  err = installLibrdkafka(extensions[idx])
+	if err != nil {
+		return err
+	}
+
+	idx = slices.IndexFunc(extensions, func(e ExtensionsData) bool { return e.Name == "mongodb" })
+  err = installMongoDB(extensions[idx], dependencyDir)
+	if err != nil {
+		return err
+	}
+
+	idx = slices.IndexFunc(extensions, func(e ExtensionsData) bool { return e.Name == "libsodium" })
+  err = installLibsodium(extensions[idx], dependencyDir)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func installRabbitMQC(extensionData ExtensionsData) error {
+	err := downloadAndExtract(
+		extensionData.Name,
+		fmt.Sprintf("https://github.com/alanxz/rabbitmq-c/archive/v%s.tar.gz", extensionData.Version),
+	)
+	if err != nil {
+		return err
+	}
+
+	dir := fmt.Sprintf("%s-%s", extensionData.Name, extensionData.Version)
+	commands := []CommandWithArgs{
+		CommandWithArgs{"cmake", []string{"."}},
+		CommandWithArgs{"cmake", []string{"build", "."}},
+		CommandWithArgs{"cmake", []string{"-DCMAKE_INSTALL_PREFIX=/usr/local", "."}},
+		CommandWithArgs{"cmake", []string{"--build", ".", "--target", "install"}},
+	}
+
+	for _, command := range commands {
+		err = runCommand(dir, command.Command, command.Args...)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func installLua(extensionData ExtensionsData) error {
+	err := downloadAndExtract(
+		extensionData.Name,
+		fmt.Sprintf("http://www.lua.org/ftp/lua-%s.tar.gz", extensionData.Version),
+	)
+	if err != nil {
+		return err
+	}
+
+	dir := fmt.Sprintf("%s-%s", extensionData.Name, extensionData.Version)
+	commands := []CommandWithArgs{
+		CommandWithArgs{"make", []string{"linux", "MYCFLAGS=-fPIC"}},
+		CommandWithArgs{"make", []string{"install"}},
+	}
+
+	for _, command := range commands {
+		err = runCommand(dir, command.Command, command.Args...)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
```

Comment:
are all of the `installX` functions the same? Can we abstract that into a separate function to cut down on duplication?

Reply:
@sophiewigmore Yes and no... some require calling `cmake` or `configure` but the options passed to those commands are different or some of the commands require using different flags.